docs/content/development/build.md

```diff
@@ -2,28 +2,70 @@
 layout: doc_page
 ---
 
-### Build from Source
-
-You can build Druid directly from source. Please note that these instructions are for building the latest stable of Druid. 
+You can build Druid directly from source. Please note that these instructions are for building the latest stable version of Druid.
 For building the latest code in master, follow the instructions [here](https://github.com/druid-io/druid/blob/master/docs/content/development/build.md).
 
+# Requirements
+
 Building Druid requires the following:
-- [JDK 8](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
-- [Maven version 3.x](http://maven.apache.org/download.cgi)
 
-To do so, run these commands:
+- [JDK 8](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)+
+- [Maven 3.x](http://maven.apache.org/download.cgi)+
+
+# Building
+
+## Build a runtime distribution
+
+Move to the project root directory and run:
+
+```
+mvn clean package -DskipTests
+```
+
+This will compile the project and create the Druid binary distribution tarball under `distribution/target/druid-VERSION-bin.tar.gz`
+and `mysql-metadata-storage` extension under `distribution/target/mysql-metadata-storage-bin.tar.gz`.
+
+- For more information on `mysql-metadata-storage` extension, see [here](../development/extensions-core/mysql.html).
+- For more information on including extensions, see [here](../operations/including-extensions.html).
+
+## Build javadoc
+
+```
+mvn install -DskipTests && mvn javadoc:javadoc
+```
+
+# Testing
+
+## Run all unit/integration tests
 
 ```
-git clone git@github.com:druid-io/druid.git
-cd druid
-mvn clean package
+export AWS_REGION=us-east-1 && mvn test -B -Pparallel-test -Dmaven.fork.count=2
```

Comment:
This looks a bit confused. First, `-Pparallel-test -Dmaven.fork.count=2` is not a mandatory option. We can add a section for running unit all unit tests in parallel. Same for other sections.

Second, `AWS_REGION=us-east-1` is not necessary if you have aws region configuration. Same for `Run s3-extensions module unit tests` section.